### PR TITLE
feature/758 balance

### PR DESCRIPTION
**Pull request by gentee**

### CODE REVIEW

File: packages/model/keys.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ func (m *Key) Get(wallet int64) (bool, error) {
 	return isFound(DBConn.Where("id = ?", wallet).First(m))
 }
 
+func (m *Key) GetTrans(transaction *DbTransaction, wallet int64) (bool, error) {
```
Comment (ghost) [inline packages/model/keys.go]: What is it? GetTransaction?

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/model/keys.go (hunk 1 above)
Comment (gentee) [reply packages/model/keys.go]: I don't know how to name this function. It does the same as Get function but in the specified transaction.

Comment (gentee) [reply packages/model/keys.go]: I have renamed it to GetWallet name.

**Review by potehinre — CHANGES_REQUESTED** [review]
It s better to add CHECK constraint on wallet amount to database schema. Like ALTER TABLE "_keys" ADD CONSTRAINT CHECK(amount > 0)

Added CHECK CONSTRAINT amount >= 0

**Review by potehinre — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**